### PR TITLE
Step Learning Rate Scheduler

### PR DESCRIPTION
**Pull request by esube**
Added a StepLearningRate scheduler and an example resnet training script for CIFAR-10 that uses it.

This is based on a recent PR that was not formed well. But, since @fchollet is interested in adding these schedulers to keras, I resurrected the PR fixing most of the issues and adding an example script that uses it in examples. This learning rate scheduler schedules a step learning rate scheduling that is common to train very deep networks such as GoogLeNet, resnets, WRN, etc...

The scheduler as implemented here could be used manually as well as automatically. If the user supplies a list of epochs, the scheduler drops the learning rate at those manually specified discrete epochs and if the user supplies a single 'patience' epoch, it will check if the monitored variable doesn't improve up to 'patience' amount of epochs and drops the learning rate by a user supplied drop amount. The auto mode is very similar to early stopping, the difference being instead of stopping this scheduler reduces the learning rate, and hence I shared a bunch of code from early stopping.

I have used the example code I shared here with n=4 and k=4 to result in WRN (wide residual network) 28x4 and reproduced (even slightly better) result with a similar implementation in Lasagne. 

Note: I had to modify the SGD optimizer code in keras to allow only user supplied scheduler like this one and not end up with sporadic 'nans' when I set lr.decay in SGD in the current implementation.

error_plots
[//]: #resnet_28x4_model

### CODE REVIEW

**Comment by esube** [comment]
I didn't run PEP8 on the code and it still has some commenting standard issues. I will fix it today. Other than that it passes the other builds.

I will also add tests if it is necessary. But, I have tested it thoroughly and I am using it in my projects.
